website/docs/pubsub.md

```diff
@@ -0,0 +1,217 @@
+
+> Redis is used for example purposes only. Dapr supports [many different state store options](https://docs.dapr.io/reference/components-reference/supported-state-stores/).
+
+#### Prerequisites
+
+1. Install Dapr
+
+    To install Dapr with specific requirements and configuration, please refer to [Dapr docs](https://docs.dapr.io/operations/hosting/kubernetes/kubernetes-deploy/).
+    > Dapr is installed with mtls enabled by default, for more details on the same plaase refer to [Dapr security](https://docs.dapr.io/operations/security/mtls/#setting-up-mtls-with-the-configuration-resource).
+
+2. Install Redis
+
+    Please refer to [this](https://docs.dapr.io/getting-started/tutorials/configure-state-pubsub/#step-1-create-a-redis-store) guide to install Redis.
+
+    > To install Redis with TLS, please refer to [this](https://docs.bitnami.com/kubernetes/infrastructure/redis-cluster/administration/enable-tls/) doc.
+
+#### Configure a sample subscriber to receive violations
+
+1. Create `fake-subscriber` namespace and redis secret
+
+```shell
+kubectl create ns fake-subscriber
+kubectl get secret redis --namespace=default -o yaml | sed 's/namespace: .*/namespace: fake-subscriber/' | kubectl apply -f - # creating redis secret in subscriber namespace to allow dapr sidecar to connect to redis instance
+```
+
+2. Create Dapr pubsub component
+```shell
+kubectl apply -f <<EOF
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: pubsub
+  namespace: fake-subscriber
+spec:
+  type: pubsub.redis
+  version: v1
+  metadata:
+  - name: redisHost
+    value: redis-master.default.svc.cluster.local:6379
+  - name: redisPassword
+    secretKeyRef: 
+      name: redis
+      key: redis-password
+```
+> Please use [this guide](https://docs.dapr.io/reference/components-reference/supported-state-stores/setup-redis/) to properly configure Redis pubsub component for Dapr.
+
+3. Deploy subscriber application
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: sub
+  namespace: fake-subscriber
+  labels:
+    app: sub
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: sub
+  template:
+    metadata:
+      labels:
+        app: sub
+      annotations:
+        dapr.io/enabled: "true"
+        dapr.io/app-id: "subscriber"
+        dapr.io/enable-api-logging: "true"
+        dapr.io/app-port: "6002"
+    spec:
+      containers:
+      - name: go-sub
+        image: fake-subscriber:latest
+        imagePullPolicy: Never
+```
+**Note:** Dockerfile to build image for fake-subscriber is under [gatekeeper/test/fake-subscriber](https://github.com/open-policy-agent/gatekeeper/tree/master/test/pubsub/fake-subscriber). You can find make rule to build and deploy subscriber in [Makefile](https://github.com/open-policy-agent/gatekeeper/blob/master/Makefile) under name `e2e-subscriber-build-load-image` and `e2e-subscriber-deploy`.
+
+#### Configure Gatekeeper with Pubsub enabled
```

Comment:
Could these steps be done with helm chart install or update by passing the required values as helm chart values parameters?

Reply:
I added command to install `gatekeeper` with helm. 

Do we want to include pubsub component that needs to be created in step 1 in our charts as well? IMO I am not sure including something for quickstart should be added to helm charts.

Reply:
> Do we want to include pubsub component that needs to be created in step 1 in our charts as well?

No